packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx

```diff
@@ -400,5 +427,33 @@ export const MetricsTable = () => {
         });
     }, [canManageTags, dataHasCategories, table]);
 
-    return <MantineReactTable table={table} />;
+    switch (metricCatalogView) {
+        case MetricCatalogView.LIST:
+            return <MantineReactTable table={table} />;
+        case MetricCatalogView.TREE:
+            return (
+                <Paper {...mantinePaperProps}>
+                    <Box>
+                        <MetricsTableTopToolbar
+                            search={search}
+                            setSearch={setSearch}
+                            totalResults={totalResults}
+                            position="apart"
+                            p={`${theme.spacing.lg} ${theme.spacing.xl}`}
+                            showCategoriesFilter={
+                                canManageTags || dataHasCategories
+                            }
+                            onMetricCatalogViewChange={handleViewChange}
+                            metricCatalogView={metricCatalogView}
+                        />
+                        <Divider color="gray.2" />
+                    </Box>
+                    <Box w="100%" h="calc(100dvh - 350px)">
+                        <MetricTree metrics={flatData} />
+                    </Box>
+                </Paper>
+            );
+        default:
+            assertUnreachable(metricCatalogView, 'Invalid metric catalog view');
+    }
```

Comment:
Added the tree as part of the table model, this way we could keep the top bar as is and reuse it (keeping the same styles without duplicating as well)